src/benchmark.cc

```diff
@@ -830,14 +862,26 @@ State::State(size_t max_iters, bool has_x, int x, bool has_y, int y,
 void State::PauseTiming() {
   // Add in time accumulated so far
   CHECK(running_benchmark);
+  CHECK(started_ && !finished_ && !error_occurred_);
   timer_manager->StopTimer();
 }
 
 void State::ResumeTiming() {
   CHECK(running_benchmark);
+  CHECK(started_ && !finished_);
   timer_manager->StartTimer();
 }
 
+void State::SkipWithError(const char* msg) {
+  CHECK(msg);
+  error_occurred_ = true;
+  const char* expected_no_error_msg = nullptr;
+  error_message.compare_exchange_weak(expected_no_error_msg, msg);
```

Comment:
I think we differ on the definition of "iteration".  Once a thread reports an error it performs no future iterations. I'll add code to enforce that each thread only calls "SkipWithError(...)" once.

This code selects the first error reported by a multi-threaded benchmark. Different threads may report an error later but IMO the first error message is likely the most relevant.